### ISSUE TITLE
pyMOR source install requires latest numpy version

### ISSUE THREAD

**Issue by sdrave** [pymor/pymor#699]
In a clean virtualenv

```sh
pip install 'numpy==1.15.4' 
pip install pymor[full] 
```

gives a

```
ValueError: numpy.ufunc size changed, may indicate binary incompatibility. Expected 216 from C header, got 192 from PyObject
```

when importing the Cython modules. Seems like the NumPy guys changed their ABI with the current 1.16 release.

Not sure what to do about it. At least we should let the wheels depend on `numpy>=1.16.0`. I guess having different wheels for different NumPy's is not an option.

**Comment by renefritze** [comment]
That command install pymor 0.5.2 though, right? Did it pull in pymess? Maybe just attach the output of `pip freeze`? 
You should not get this with a master wheel. https://zivgitlab.uni-muenster.de/pymor/pymor/-/jobs/158676/artifacts/file/shared/pymor-999!1+2019.2rc0.388.g0d317e1a-cp36-cp36m-manylinux1_x86_64.whl

**Comment by sdrave** [comment]
Same error with

```
pip install 'numpy==1.15.4' 
pip install git+https://github.com/pymor/pymor#egg=pymor[full]
```

and 
```python
import pymor.tools.relations
```

which is really strange as the module is compiled from source. pymess is not installed.

**Comment by renefritze** [comment]
I just checked and the manylinux wheels our CI produces are indeed fine. 

I have no idea what the real issue is here, as it really should not happen in the source-to-wheel install. 

BTW, this is a fine example of something https://github.com/pymor/pymor/pull/646 would catch.

**Comment by sdrave** [comment]
Can you reproduce the issue?

**Comment by renefritze** [comment]
Yeah. With older cython too.

**Comment by sdrave** [comment]
> With older cython too.

What exactly do you mean by that? You tested `numpy==1.15.4` with an older cython version?

**Comment by sdrave** [comment]
Ok, I did some more testing: As it seems, `Cython` is not the issue. I could build a usable `relations.so` with latest `Cython` by using `pip install -e /path/to/pymor` and `python setup.py build_ext --inplace`. So I guess the culprit might be:

```
Building wheel for pymor (PEP 517) ... done
```

when building pyMOR from source?

**Comment by renefritze** [comment]
> What exactly do you mean by that? You tested `numpy==1.15.4` with an older cython version?

Yes


> So I guess the culprit might be:
> 
> ```
> Building wheel for pymor (PEP 517) ... done
> ```
> 
> when building pyMOR from source?

Right. PEP 517 basically says to create this wheel in a 'clean' environment with only the dependencies as specified in the `pyproject.toml`.  That wheel must then carry the wrong numpy dependency, otherwise installing the wheel would bump the numpy installed in the original venv too. 
I'll look into this, but I see no immediate good solution.

Have you been using `pip==19.1.1`, @sdrave ?

**Comment by sdrave** [comment]
Yeah, this was with `pip==19.1.1`. So this basically means that we need to pin a specific numpy version (class of versions with same ABI) to be able to install pyMOR reliably from source? This is totally stupid.

**Comment by sdrave** [comment]
Regarding the changed title: I also get the error with the latest pyMOR wheel from wheels.pymor.org. I guess, with wheels there is no way around this issue as these can only work with one ABI. So at least the wheels would need to pin a class of ABI compatible numpy versions.

**Comment by renefritze** [comment]
This will be fixed with #708
